azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/MainActivity.kt

```diff
@@ -21,58 +21,55 @@ import androidx.activity.result.contract.ActivityResultContracts
 import androidx.activity.viewModels
 import androidx.appcompat.app.AppCompatActivity
 import androidx.core.content.ContextCompat
-import androidx.fragment.app.FragmentFactory
 import androidx.fragment.app.FragmentTransaction
 import androidx.lifecycle.lifecycleScope
 import com.azure.android.communication.ui.R
 import com.azure.android.communication.ui.configuration.CallCompositeConfiguration
-import com.azure.android.communication.ui.di.DIContainer
-import com.azure.android.communication.ui.error.ErrorHandler
 import com.azure.android.communication.ui.presentation.fragment.calling.CallingFragment
 import com.azure.android.communication.ui.presentation.fragment.setup.SetupFragment
-import com.azure.android.communication.ui.presentation.manager.AudioSessionManager
-import com.azure.android.communication.ui.presentation.manager.LifecycleManager
-import com.azure.android.communication.ui.presentation.manager.PermissionManager
 import com.azure.android.communication.ui.presentation.navigation.BackNavigation
-import com.azure.android.communication.ui.presentation.navigation.NavigationRouter
-import com.azure.android.communication.ui.redux.Store
 import com.azure.android.communication.ui.redux.action.CallingAction
-import com.azure.android.communication.ui.redux.middleware.handler.CallingMiddlewareActionHandler
 import com.azure.android.communication.ui.redux.state.NavigationStatus
-import com.azure.android.communication.ui.redux.state.ReduxState
 import com.azure.android.communication.ui.service.calling.InCallService
 import kotlinx.coroutines.launch
 
 internal class MainActivity : AppCompatActivity() {
 
     private val diContainerHolder: DependencyInjectionContainerHolder by viewModels()
-
-    private lateinit var navigationRouter: NavigationRouter
-    private lateinit var fragmentFactory: FragmentFactory
-    private lateinit var store: Store<ReduxState>
-    private lateinit var configuration: CallCompositeConfiguration
-    private lateinit var permissionManager: PermissionManager
-    private lateinit var audioSessionManager: AudioSessionManager
-    private lateinit var lifecycleManager: LifecycleManager
-    private lateinit var errorHandler: ErrorHandler
-    private lateinit var callingMiddlewareActionHandler: CallingMiddlewareActionHandler
-    private lateinit var videoViewManager: VideoViewManager
+    private val container by lazy { diContainerHolder.container }
+
```

Comment:
Using the get() syntax is better than fields, since it's more of a "reference" kind of thing to something already stored in the container. This is just syntactic sugar.